src/main/java/spireMapOverhaul/zones/candyland/CandyLand.java

```diff
@@ -0,0 +1,197 @@
+            cards.add(c);
+        }
+    }
+
+    private boolean containCard(ArrayList<AbstractCard> list, AbstractCard c){
+        for(AbstractCard card : list)
+            if (Objects.equals(c.cardID, card.cardID))
+                return true;
+        return false;
+    }
+
+    @Override
+    public void modifyReward(RewardItem rewardItem) {
+        if(rewardItem.type == RewardItem.RewardType.RELIC && !(rewardItem.relic instanceof Mango || rewardItem.relic instanceof Pear || rewardItem.relic instanceof Strawberry || rewardItem.relic instanceof Waffle)){
+            switch(rewardItem.relic.tier){
+                case RARE:
+                    SpireAnniversary6Mod.logger.info("Replacing " + rewardItem.relic.name + " with Mango");
+                    rewardItem.relic = new Mango();
+                    rewardItem.text = rewardItem.relic.name;
+                    break;
+                case UNCOMMON:
+                    SpireAnniversary6Mod.logger.info("Replacing " + rewardItem.relic.name + " with Pear");
+                    rewardItem.relic = new Pear();
+                    rewardItem.text = rewardItem.relic.name;
+                    break;
+                case COMMON:
+                    SpireAnniversary6Mod.logger.info("Replacing " + rewardItem.relic.name + " with Strawberry");
+                    rewardItem.relic = new Strawberry();
+                    rewardItem.text = rewardItem.relic.name;
+                    break;
+                default:
+                    SpireAnniversary6Mod.logger.info("Non-standard rarity found, replacing " + rewardItem.relic.name + " with Waffle");
+                    rewardItem.relic = new Waffle();
+                    rewardItem.text = rewardItem.relic.name;
+                    break;
+            }
+        } else if(rewardItem.type == RewardItem.RewardType.POTION){
+            while(rewardItem.potion.isThrown){
+                SpireAnniversary6Mod.logger.info("Try to replace " + rewardItem.potion.name + "with drinkable potion.");
+                rewardItem.potion = AbstractDungeon.returnRandomPotion(rewardItem.potion.rarity, false);
+            }
+        }
+    }
+
+    @Override
+    public void modifyRewards(ArrayList<RewardItem> rewards) {
+        SpireAnniversary6Mod.logger.info("Turning Gold into Chocolate");
+        for(RewardItem r : (ArrayList<RewardItem>) rewards.clone()){
+            if(r.type == RewardItem.RewardType.GOLD){
+                rewards.add(new HealReward("ChocolateCoins.png", r.goldAmt));
+            }
+        }
+        rewards.removeIf(r -> r.type == RewardItem.RewardType.GOLD);
+    }
+
+    public void postAddButtons(ArrayList<AbstractCampfireOption> buttons) {
+        for (AbstractCampfireOption button : buttons){
+            if (button instanceof SmithOption){
+                button.usable = false;
+            }
+        }
+    }
+
+    public void postUseCampfireOption(AbstractCampfireOption option) {
```

Comment:
You should change the rest campfire option text to say that it does this..

I don't think we have any hook for doing that, but it should be easy to patch. You can do it with a postfix patch for the constructor of `RestOption` that adds on to the `description` field.

Reply:
Actually, it turns out you can use postAddButtons to get the button for RestOption, then use reflection hacks to get and set the description. I'll update the documentation to mention changing descriptions when using postUseCampfireOption.